src/config.coffee

```diff
@@ -677,6 +677,34 @@ class Config
     finally
       @endTransaction()
 
+  ###
+  Section: Internal methods used by core
+  ###
+
+  # Private: Suppress calls to handler functions registered with {::onDidChange}
+  # and {::observe} for the duration of the {Promise} returned by `callback`.
+  # After the {Promise} is either resolved or rejected, handlers will be called
+  # once if the value for their key-path has changed.
+  #
+  # * `callback` {Function} that returns a {Promise}, which will be executed
+  #   while suppressing calls to handlers.
+  #
+  # Returns a {Promise} that is either resolved or rejected according to the
+  # `{Promise}` returned by `callback`. If `callback` throws an error, a
+  # rejected {Promise} will be returned instead.
+  transactAsync: (callback) ->
+    @beginTransaction()
+    try
+      endTransaction = (fn) => (args...) =>
```

Comment:
Yeah, I guess so: this is a function that returns a function (i.e. curried), so when called with `endTransaction(resolve)`, for instance, will return a function that:

* Ends the transaction.
* Calls `resolve` with all the passed arguments.

When used like `result.then(endTransaction(resolve))`, `resolve` will be invoked with all the arguments available in `then`'s callback. This means that a potential client could use this API like the following:

```coffee
atom.config.transactAsync(-> Promise.resolve(1)).then((n) -> console.log("Output: #{n}"))
#=> Output: 1
```

Although we're not using it in our production code, it seemed reasonable to follow the `Promise` API semantics when I implemented this. What do you think?